// this file describes protocol buffer messages sent by the client when
// communicating with a pblog server

package pblog.protocol.request;

// requesting status of the pblog server's store
message StoreInfo {

}

// start streaming events as they hit the remote store
// this is not reliable
message StartStream {
    // list of paths to subscribe to
    // string value is a store path (e.g. '/bucket/set')
    repeated string paths = 1;
}

// download stream(s) from the server
message Get {
    // list of streams to download
    repeated string paths = 1;
    
    // list of offsets within the remote stream at which to start the download
    repeated uint64 stream_offsets = 2;
}

